dataactvalidator/config/sqlrules/c9_award_financial.sql

```diff
@@ -1,14 +1,22 @@
-SELECT afa.row_number, afa.fain, afa.uri
+SELECT DISTINCT afa.row_number, afa.fain, afa.uri
 FROM award_financial_assistance AS afa
 WHERE afa.submission_id = {0}
-	AND CAST(afa.federal_action_obligation as numeric) > 0
-	AND afa.row_number NOT IN (
-		SELECT afa.row_number
-		FROM award_financial_assistance AS afa
-			JOIN award_financial AS af
-				ON (afa.fain IS NOT DISTINCT FROM af.fain
-				AND afa.uri IS NOT DISTINCT FROM af.uri
-				AND afa.submission_id = af.submission_id)
-		WHERE CAST(afa.federal_action_obligation as numeric) > 0
-			AND afa.submission_id = {0}
-	);
\ No newline at end of file
+    AND COALESCE(CAST(afa.federal_action_obligation as numeric),0) <> 0
+    AND COALESCE(CAST(afa.original_loan_subsidy_cost as numeric),0) <> 0
+    AND (
+        afa.row_number NOT IN (
+            SELECT afa.row_number
+            FROM award_financial_assistance AS afa
+                JOIN award_financial AS af
+                    ON (afa.submission_id = af.submission_id
+                        AND (afa.fain IS NOT DISTINCT FROM af.fain
+                            AND afa.uri IS NOT DISTINCT FROM af.uri)
+                    )
+            WHERE COALESCE(CAST(afa.federal_action_obligation as numeric),0) <> 0
+                AND COALESCE(CAST(afa.original_loan_subsidy_cost as numeric),0) <> 0
+                AND afa.submission_id = {0}
+        ) OR (
+            afa.fain IS NOT NULL
+            AND afa.uri IS NOT NULL
```

Comment:
So after seeing it in the unit test below I realized this check is being done on D2, which we should definitely confirm as I thought the expectation was that D2 can have both and file C should be just one